File: Environment.py
import datetime
import random
from time_manager import TimeManager

class Pet:
    def __init__(self, name):
        self.name = name
        self.thirst = 0
        self.age = 0
        self.mood = "Happy"
        self.level = 1
        self.hp = 100
        self.isSick = False
        self.status = "Alive"

class Environment:
    def __init__(self, pet):
        self.pet = pet
        self.time_manager = TimeManager()

    def update_pet_status(self):
        elapsed_time = self.time_manager.get_elapsed_time().total_seconds()

        # Frissítjük a kisállat tulajdonságait az eltelt idő alapján
        self.pet.thirst += elapsed_time * 0.02
        self.pet.age += elapsed_time / 86400 
        self.pet.hp -= (5 if self.pet.isSick else 1) * (elapsed_time * 0.01)

        # állapotok beállítása
        self.pet.thirst = min(100, self.pet.thirst)
        self.pet.hp = max(0, self.pet.hp)
        self.pet.status = "Dead" if self.pet.hp == 0 else "Alive"

        self.pet.mood = "Grumpy" if self.pet.thirst > 50 else "Happy"

    def random_sickness_event(self):
        # Véletlenszerűen betegség állapotának beállítása
        self.pet.isSick = random.choice([True, False])

File: loligochi/loligochi-classlib/environment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace loligochi_classlib
{
    internal class environment
    {
        public class Pet
        {
            public string Name { get; set; }
            public double Thirst { get; set; }
            public double Age { get; set; }
            public string Mood { get; set; }
            public int Level { get; set; }
            public double Hp { get; set; }
            public bool IsSick { get; set; }
            public string Status { get; set; }

            public Pet(string name)
            {
                Name = name;
                Thirst = 0;
                Age = 0;
                Mood = "Happy";
                Level = 1;
                Hp = 100;
                IsSick = false;
                Status = "Alive";
            }
        }

        public class Environment
        {
            private Pet Pet { get; set; }
            private time_manager TimeManager { get; set; }
            private Random RandomGenerator { get; set; }

            public Environment(Pet pet)
            {
                Pet = pet;
                TimeManager = new time_manager();
                RandomGenerator = new Random();
            }

            public void UpdatePetStatus()
            {
                var elapsedTime = TimeManager.GetElapsedTime().TotalSeconds;

                // Update pet's attributes based on elapsed time
                Pet.Thirst += elapsedTime * 0.02;
                Pet.Age += elapsedTime / 86400;
                Pet.Hp -= (Pet.IsSick ? 5 : 1) * (elapsedTime * 0.01);

                // Set conditions
                Pet.Thirst = Math.Min(100, Pet.Thirst);
                Pet.Hp = Math.Max(0, Pet.Hp);
                Pet.Status = Pet.Hp == 0 ? "Dead" : "Alive";

                Pet.Mood = Pet.Thirst > 50 ? "Grumpy" : "Happy";
            }

            public void RandomSicknessEvent()
            {
                // Randomly set sickness status
                Pet.IsSick = RandomGenerator.Next(2) == 0;
            }
        }
    }
}
